src/test/java/rx/SingleTest.java

```diff
@@ -1353,4 +1386,364 @@ public String call() throws Exception {
+            this.s = s;
+        }
+
+        /* used to simulate subscription */
+        public void sendOnCompleted() {
+            observer.onCompleted();
+        }
+
+        /* used to simulate subscription */
+        public void sendOnNext(String value) {
+            observer.onNext(value);
+        }
+
+        /* used to simulate subscription */
+        public void sendOnError(Throwable e) {
+            observer.onError(e);
+        }
+
+        @Override
+        public void call(Subscriber<? super String> observer) {
+            this.observer = observer;
+            observer.add(s);
+        }
+    }
+
+    private static class TestSingle implements Single.OnSubscribe<String> {
+
+        SingleSubscriber<? super String> subscriber = null;
+        Subscription s;
+
+        public TestSingle(Subscription s) {
+            this.s = s;
+        }
+
+        /* used to simulate subscription */
+        public void sendOnSuccess(String value) {
+            subscriber.onSuccess(value);
+        }
+
+        /* used to simulate subscription */
+        public void sendOnError(Throwable e) {
+            subscriber.onError(e);
+        }
+
+        @Override
+        public void call(SingleSubscriber<? super String> observer) {
+            this.subscriber = observer;
+            observer.add(s);
+        }
+    }
+
+    @Test
+    public void testUntilFires() {
+        PublishSubject<Integer> source = PublishSubject.create();
+        PublishSubject<Integer> until = PublishSubject.create();
+
+        TestSubscriber<Integer> ts = new TestSubscriber<Integer>();
+
+        source.take(1).toSingle().takeUntil(until.take(1).toSingle()).unsafeSubscribe(ts);
+
+        assertTrue(source.hasObservers());
+        assertTrue(until.hasObservers());
+
+        until.onNext(1);
+
+        ts.assertError(CancellationException.class);
+        ts.assertTerminalEvent();
+
+        assertFalse("Source still has observers", source.hasObservers());
+        assertFalse("Until still has observers", until.hasObservers());
+        assertFalse("TestSubscriber is unsubscribed", ts.isUnsubscribed());
+    }
+
+    @Test
+    public void testUntilFiresObservable() {
+        PublishSubject<Integer> source = PublishSubject.create();
+        PublishSubject<Integer> until = PublishSubject.create();
+
+        TestSubscriber<Integer> ts = new TestSubscriber<Integer>();
+
+        source.take(1).toSingle().takeUntil(until.take(1)).unsafeSubscribe(ts);
+
+        assertTrue(source.hasObservers());
+        assertTrue(until.hasObservers());
+
+        until.onNext(1);
+
+        ts.assertError(CancellationException.class);
+        ts.assertTerminalEvent();
+
+        assertFalse("Source still has observers", source.hasObservers());
+        assertFalse("Until still has observers", until.hasObservers());
+        assertFalse("TestSubscriber is unsubscribed", ts.isUnsubscribed());
+    }
+
+    @Test
+    public void testDownstreamUnsubscribes() {
```

Comment:
Since you're adding tests to `SingleTest` you should definitely mention target method (operator) name in the test